src/inmanta/deploy/scheduler.py

```diff
@@ -385,14 +385,14 @@ async def _new_version(
             async with self._scheduler_lock:
                 self._state.version = version
                 for resource in blocked_resources:
-                    self._state.block_resource(resource, details, transient=False)
+                    self._state.block_resource(resource, resources[resource], is_transitive=False)
                 for resource in new_desired_state:
                     self._state.update_desired_state(resource, resources[resource])
                 for resource in added_requires.keys() | dropped_requires.keys():
                     self._state.update_requires(resource, requires[resource])
-                transitively_blocked_resources: set[ResourceIdStr] = self._state.block_provides(resources=blocked_resources)
+                transitively_blocked_resources: Set[ResourceIdStr] = self._state.block_provides(resources=blocked_resources)
                 for resource in unblocked_resources:
-                    self._state.unblock_resource(resource)
+                    self._state.mark_as_defined(resource, resources[resource])
```

Comment:
Why not simply add these resources to `new_desired_state` as well? It's semantically accurate and it prevents having a third place where we update resource intent.

It does have the disadvantage that it adds it to `dirty`, while that may only be appropriate if it's not transitively blocked. But the same actually goes for any resource who's desired state is updated while it's transitively blocked, so we should probably simply check if a resource blocked prior to marking it dirty in `update_desired_state`?

Reply:
For future reference: the first concern was deemed irrelevant in the current status of the ticket, though it will likely become relevant in the context of follow-up bugfix tickets, somewhat related to the other two points. I will create relevant tickets and link them here.

Reply:
#8284